src/storage/admin/AdminTaskManager.h

```diff
@@ -61,6 +71,8 @@ class AdminTaskManager {
   TaskContainer tasks_;
   TaskQueue taskQueue_;
   std::unique_ptr<thread::GenericWorker> bgThread_;
+  storage::StorageEnv* env_{nullptr};
+  std::unique_ptr<std::thread> unreportedAdminThread_;
```

Comment:
Where to join the thread? And is it necessary to add a new thread?

Reply:
If join the thread in the destructor or somewhere,it is possible to be in  infinite loop when metad is shut down.So I think just destruct the thread without join is ok.
The reason of adding a new thread is that we need a asynchronous way to handle the unreported tasks, or it's possible to make storaged's bootstrap too long.